src/weights.jl

```diff
@@ -569,47 +566,47 @@ function quantile(v::RealVector{V}, w::AbstractWeights{W}, p::RealVector) where
         x < 0 && error("weight vector cannot contain negative entries")
     end
 
-    # full sort
-    vw = sort!(collect(zip(v, w.values)))
 
-    wsum = w.sum
+    #remove zeros weights and sort
+    wsum = sum(w.value)
```

Comment:
Use `sum(w)`, which is equivalent to ` w.sum` thanks to a special method. I don't think you need to access the `values` field below either: `AbstractWeight` implements the `AbstractVector` interface (but does not guarantee that the `values` field exists).